src/lsp/lsp_base.ml

```diff
@@ -41,15 +41,24 @@ let mk_range (p : Pos.pos) : J.t =
 let json_of_goal g =
   let pr_hyp (s,(_,t,_)) =
     let t = Format.asprintf "%a" Print.pp_term (Bindlib.unbox t) in
-    `Assoc ["hname", `String s; "htype", `String t] in
+    `Assoc ["hname", `String s; "htype", `String t]
+  in
   let open Proof in
-  let g_meta = Goal.get_meta g in
-  let hyp, typ = Goal.get_type g in
-  let j_env = List.map pr_hyp hyp in
-  `Assoc [
-    "gid", `Int g_meta.meta_key
-  ; "hyps", `List j_env
-  ; "type", `String (Format.asprintf "%a" Print.pp_term typ)]
+  match g with
+  | Goal.Typ g ->
+    let g_meta = Goal.get_meta g in
+    let hyp, typ = Goal.get_type g in
+    let j_env = List.map pr_hyp hyp in
+    `Assoc [
+      "typeofgoal", `String "Typ "
+    ; "gid", `Int g_meta.meta_key
+    ; "hyps", `List j_env
+    ; "type", `String (Format.asprintf "%a" Print.pp_term typ)]
+  | Goal.Unif (ctx,t1,t2) ->
+    let constr = Format.asprintf "%a" Print.pp_constr (ctx,t1,t2) in
+    `Assoc [
+      "typeofgoal", `String "Unif"
+    ; "constr", `String constr]
```

Comment:
This needs update:
``` 
export interface Goal {
	gid :  number // goal id
	hyps : Env[] // hypothesis
	type : String // goals
}
```
I'm not an expert on TS, but should be easy. I won't do it this week tho as I'm pretty busy with CUDW.